NRLMMD-GEOIPS/geoips
# Move PyTest functionality from PluginRegistry Class to Unit Tests

# Reviewer Checklist


# Related Issues
fixes NRLMMD-GEOIPS/geoips#426

# Testing Instructions
To run the new unit tests, simply run ``pytest test_plugin_registries.py -v``. This will unit test the plugin registry and give you verbose output of what occurs. All passes should test other than three, which fail (expected failure). This file can be found at geoips/tests/unit_tests/plugin_registries/test_plugin_registries.py.

# Summary
*From GEOIPS#426: 2024-02-02, GeoIPS won't run pip install . w/out [test] target*

GeoIPS currently fails to run if installed by simply calling ``pip install .``. It must,
instead, be installed with pip install .[test]. This is because pytest is used in
geoips/plugin_registry.py but is not included in the base GeoIPS installation
requirements.

To remediate this, we've since moved the pytest validation functionality off of the
PluginRegistry class itself, and moved it into unit tests at
``tests/unit_tests/plugin_registries/test_plugin_registries.py``, under class
PluginRegistryValidator. This removes the need for PyTest to be a GeoIPS dependency,
and is now a Unit Test dependency. If users want to validate their plugin registry, they
will have to manually install pytest to validate from now on.

    - modified: geoips/plugin_registry.py
    - modified: tests/unit_tests/plugin_registries/test_plugin_registries.py

# Output
Shown below is the expected result of running the pytest command listed above.

![Screenshot 2024-02-16 at 11 41 01 AM](https://github.com/NRLMMD-GEOIPS/geoips/assets/62626811/ec60c53f-c1d6-4634-bb1a-9c30aabeb65c)
